### ISSUE TITLE
Exclude waterloggable blocks from water MLG

### ISSUE THREAD

**Issue by maximilian-hack** [cabaletta/baritone#4548]
## Describe your suggestion
When testing with the Baritone API (baritone-api-fabric-1.11.0 for 1.21.1; compiled by following your guide) I found out, that it does not seem to care wether the block it tries to land a water mlg on is waterloggable. This may result in falldamage which is not the desired outcome.
In my opinion there should be an exclusion list for waterloggable blocks.

## Settings
Add an exclusion list for not MLG-able blocks. This list should include upper slabs and simliar blocks.

## Context
This would make the Water MLG more reliable and less dangerous.

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by ZacSharp** [comment]
Kind of a duplicate of #3950 

It's crazy how much work for Baritone I already started and abandoned.

**Comment by boasbakker** [comment]
Yeah this issue is still quite annoying. It also goes wrong when there are stairs which already waterlogged, Baritone will consider those safe to fall onto but they are not safe. Should I make a separate issue for this?

**Comment by boasbakker** [comment]
Isn't this more a bug than a feature request?
